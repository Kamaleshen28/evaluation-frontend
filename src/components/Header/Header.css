.header-wrapper {
  background-color: black;
  color: white;
  height: 100px;
  display: flex;
  align-items: center;
  padding: 0 240px 0 240px;
  font-size: 50px;
}