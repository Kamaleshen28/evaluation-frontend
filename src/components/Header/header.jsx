import React from 'react';
import { useNavigate } from 'react-router-dom';
import './Header.css';

export default function Header() {

  const navigate = useNavigate();
  const handleNavigation = () => {
    navigate('/');
  };
  return (
    <div className="header">
      <div className="header-wrapper">
        <span className="header-title" onClick={handleNavigation} data-testid="evenity-text">EVENTIFY</span>
      </div>
    </div>
  );
}