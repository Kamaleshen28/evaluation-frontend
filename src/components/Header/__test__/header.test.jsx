import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import Header from '../header';
import { useNavigate } from 'react-router-dom';


jest.mock('react-router-dom');


describe('card', () => {
  it('should render header correctly', async () => {
    const { asFragment } = render(<Header />);
    await waitFor(() => {
      expect(screen.getAllByText('EVENTIFY')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
  it('should call naviagte with / when EVENTIFY is clicked', async () => {
    const navigateFn = jest.fn();
    useNavigate.mockReturnValue(navigateFn);
    render(<Header />);
    const Evenity = screen.getByTestId('evenity-text');
    fireEvent.click(Evenity);
    expect(navigateFn).toBeCalledWith('/');
  });
});



