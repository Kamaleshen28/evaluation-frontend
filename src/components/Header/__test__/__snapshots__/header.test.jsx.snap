// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render header correctly 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header-wrapper"
    >
      <span
        class="header-title"
        data-testid="evenity-text"
      >
        EVENTIFY
      </span>
    </div>
  </div>
</DocumentFragment>
`;
